src/libsystemd/sd-device/device-monitor.c

```diff
@@ -718,41 +718,30 @@ _public_ int sd_device_monitor_filter_update(sd_device_monitor *m) {
 }
 
 _public_ int sd_device_monitor_filter_add_match_subsystem_devtype(sd_device_monitor *m, const char *subsystem, const char *devtype) {
-        _cleanup_free_ char *s = NULL, *d = NULL;
         int r;
 
         assert_return(m, -EINVAL);
         assert_return(subsystem, -EINVAL);
 
-        s = strdup(subsystem);
-        if (!s)
-                return -ENOMEM;
-
-        if (devtype) {
-                d = strdup(devtype);
-                if (!d)
-                        return -ENOMEM;
-        }
-
-        r = hashmap_ensure_put(&m->subsystem_filter, NULL, s, d);
-        if (r < 0)
+        r = hashmap_put_strdup_full(&m->subsystem_filter, &trivial_hash_ops_free_free, subsystem, devtype);
```

Comment:
shouldn't this be string_hash_ops_free_free?

Reply:
No, it must be `trivial_hash_ops_free_free`. As user may set two or more different devtypes for the same subsystem. See #17259 and a0887abbd8bd9f1a9a975af08e6b4a43960bb3e2.

Reply:
could you add a comment about this here?

Reply:
Sure, added.